### PR TITLE
Introducing debugger modes (run, breakpoint).

### PR DESCRIPTION
**Pull request by zherczeg**

### CODE REVIEW

**Comment by yichoi** [comment]
LGTM. Good Start!

File: jerry-core/debugger/jerry-debugger.h [hunk 1]
```diff
@@ -82,12 +107,17 @@ typedef enum
   JERRY_DEBUGGER_EVAL_ERROR_END = 20, /**< last part of eval result when an error is occured */
 
   /* Messages sent by the client to server. */
+    /* The following messages are accepted in both run and breakpoint modes. */
```
Comment (LaszloLango) [inline jerry-core/debugger/jerry-debugger.h]: wrong indentation

File: jerry-core/debugger/jerry-debugger.h [hunk 2]
```diff
@@ -82,12 +107,17 @@ typedef enum
...
   JERRY_DEBUGGER_FREE_BYTE_CODE_CP = 1, /**< free byte code compressed pointer */
   JERRY_DEBUGGER_UPDATE_BREAKPOINT = 2, /**< update breakpoint status */
   JERRY_DEBUGGER_STOP = 3, /**< stop execution */
+    /* The following messages are only available in breakpoint
```
Comment (LaszloLango) [inline jerry-core/debugger/jerry-debugger.h]: wrong indentation

File: jerry-core/debugger/jerry-debugger.h [hunk 3]
```diff
@@ -82,12 +107,17 @@ typedef enum
...
+     * mode and they switch the engine to run mode. */
   JERRY_DEBUGGER_CONTINUE = 4, /**< continue execution */
   JERRY_DEBUGGER_STEP = 5, /**< next breakpoint, step into functions */
   JERRY_DEBUGGER_NEXT = 6, /**< next breakpoint in the same context */
+    /* The following messages are only available in breakpoint
```
Comment (LaszloLango) [inline jerry-core/debugger/jerry-debugger.h]: wrong indentation

File: jerry-core/debugger/jerry-debugger.h (hunk 1 above)
Comment (zherczeg) [inline jerry-core/debugger/jerry-debugger.h]: This indentation was intended from my side. Perhaps I could bring the `/*`one space earlier, and add a white space after it (so there will be two spaces after the `/*`). What do you think?

Comment (LaszloLango) [inline jerry-core/debugger/jerry-debugger.h]: That would be worse. Leave as is then.

Comment (zherczeg) [inline jerry-core/debugger/jerry-debugger.h]: What about adding two spaces before `/*`?

Comment (tilmannOSG) [inline jerry-core/debugger/jerry-debugger.h]: Why indent the comments in the first place? Not indenting the comments and adding blank lines to separate the three different categories would seem the best way to highlight this IMHO.

Comment (zherczeg) [inline jerry-core/debugger/jerry-debugger.h]: I didn't like that, but that is an option. If @LaszloLango likes it then I can do it.

Comment (LaszloLango) [inline jerry-core/debugger/jerry-debugger.h]: I don't have strong preferences. Those are only comments. :) Their content is the important part. I've already approved the PR.

**Pull request merged**